src/static_handler.cc

```diff
@@ -0,0 +1,82 @@
+#include <sstream>
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/fstream.hpp>
+#include "not_found_handler.h"
+#include "static_handler.h"
+
+RequestHandler::Status StaticHandler::Init(const std::string& uri_prefix,
+                                           const NginxConfig& config) {
+  uri_prefix_ = uri_prefix;
+  RequestHandler::ParseConfig(config);
+  root_ = config_map_[CONFIG_ROOT_KEY];
+  return OK;
+}
+
+// todo: add request validation
+RequestHandler::Status StaticHandler::HandleRequest(const Request& request,
+                                                    Response* response) {
+  if (response == nullptr) {
+    return ERROR;
+  }
+
+  // todo: change how we get relative_path
```

Comment:
..? Is this currently looking for a uri prefix in static handler that matches the map, and if it doesnt, you reduce your size by 1?